python/event_generator.cpp

```diff
@@ -64,27 +72,29 @@ struct regular_schedule_shim {
 struct explicit_schedule_shim {
     using time_type = arb::time_type;
 
-    std::list<time_type> times;
-
-    explicit_schedule_shim(std::list<time_type> l) {
-        for (auto& t: l) {
-            times.push_back(pyarb::assert_predicate(t, is_nonneg, "time must be a non-negative number."));
-        }
+    std::vector<time_type> times;
 
-        // Sort the times in ascending order if necessary
-        if (!std::is_sorted(times.begin(), times.end())) {
-            times.sort();
-        }
+    explicit_schedule_shim() = default;
 
+    explicit_schedule_shim(std::vector<time_type> t) {
+        set_times(t);
     }
 
     // getter and setter (in order to assert when being set)
-    void set_times(const std::list<time_type> t0) {
-        for (auto& t: t0) {
-            times.push_back(pyarb::assert_predicate(t, is_nonneg, "time must be a non-negative number."));
+    void set_times(std::vector<time_type> t) {
```

Comment:
use `std::vector` instead of `std::list`, because it is faster. Particularly applying `sort`.